applications/GeoMechanicsApplication/custom_elements/transient_Pw_line_element.h

```diff
@@ -63,34 +71,72 @@ class KRATOS_API(GEO_MECHANICS_APPLICATION) TransientPwLineElement : public Elem
         rResult = Geo::DofUtilities::ExtractEquationIdsFrom(GetDofs());
     }
 
+    void Initialize(const ProcessInfo& rCurrentProcessInfo) override
+    {
+        if (const std::size_t number_integration_points =
+                GetGeometry().IntegrationPointsNumber(GetIntegrationMethod());
+            mRetentionLawVector.size() != number_integration_points) {
+            mRetentionLawVector.resize(number_integration_points);
+        }
+        for (unsigned int i = 0; i < mRetentionLawVector.size(); ++i) {
+            mRetentionLawVector[i] = RetentionLawFactory::Clone(GetProperties());
+            mRetentionLawVector[i]->InitializeMaterial(
+                GetProperties(), GetGeometry(),
+                row(GetGeometry().ShapeFunctionsValues(GetIntegrationMethod()), i));
+        }
+    }
+
+    void InitializeSolutionStep(const ProcessInfo&) override
+    {
+        RetentionLaw::Parameters RetentionParameters(this->GetProperties());
+        for (const auto& retention_law : mRetentionLawVector) {
+            retention_law->InitializeSolutionStep(RetentionParameters);
+        }
+    }
+
+    void FinalizeSolutionStep(const ProcessInfo&) override
+    {
+        RetentionLaw::Parameters RetentionParameters(this->GetProperties());
+        for (const auto& retention_law : mRetentionLawVector) {
+            retention_law->FinalizeSolutionStep(RetentionParameters);
+        }
+    }
```

Comment:
If Initialize and FinalizeSolutionStep should be done for retention laws, maybe the same is necessary for constitutive laws.